src/modules/noit_lua_noit_binding.c

```diff
@@ -126,12 +142,227 @@ lua_noit_checks_unsubscribe(lua_State *L) {
 }
 
 static int
+noit_metric_value_index_func(lua_State *L) {
+  int n;
+  const char *k;
+  metric_value_t **udata, *metric_id;
+  n = lua_gettop(L); /* number of arguments */
+  assert(n == 2);
+  if(!luaL_checkudata(L, 1, "metric_value_t")) {
+    luaL_error(L, "metatable error, arg1 not a metric_value_t!");
+  }
+  udata = lua_touserdata(L, 1);
+  metric_id = *udata;
+  if(!lua_isstring(L, 2)) {
+    luaL_error(L, "metatable error, arg2 not a string!");
+  }
+
+  switch (*k) {
+    case 'w':
+      k = lua_tostring(L, 2);
+      if(!strcmp(k, "whence_ms")) {
+        lua_pushinteger(L, metric_id->whence_ms / 1000);
+      } else {
+        break;
+      }
+      return 1;
+    case 't':
+      if(!strcmp(k, "type")) {
+        lua_pushinteger(L, metric_id->type);
+      } else {
+        break;
+      }
+      return 1;
+    case 'v':
+      if(!strcmp(k, "value")) {
+        switch (metric_id->type) {
+          case METRIC_INT32:
+            lua_pushinteger(L, metric_id->value.v_int32);
+            break;
+          case METRIC_INT64:
+            lua_pushinteger(L, metric_id->value.v_int64);
+            break;
+          case METRIC_UINT32:
+            lua_pushinteger(L, metric_id->value.v_int32);
+            break;
+          case METRIC_UINT64:
+            lua_pushinteger(L, metric_id->value.v_int64);
+            break;
+          case METRIC_DOUBLE:
+            lua_pushnumber(L, metric_id->value.v_double);
+            break;
+          case METRIC_STRING:
+            lua_pushstring(L, metric_id->value.v_string);
+            break;
+        }
+        return 1;
+      }
+      break;
+    default:
+      break;
+  }
+  luaL_error(L, "metric_value_t no such element: %s", k);
+  return 0;
+}
+
+static void
+noit_lua_setup_metric_value(lua_State *L,
+    metric_value_t *value) {
+  metric_value_t **addr;
+  addr = (metric_value_t **)lua_newuserdata(L, sizeof(value));
+  *addr = value;
+  if(luaL_newmetatable(L, "metric_value_t") == 1) {
+    lua_pushcclosure(L, noit_metric_value_index_func, 0);
+    lua_setfield(L, -2, "__index");
+  }
+  lua_setmetatable(L, -2);
+}
+
+static int noit_metric_id_index_func(lua_State *L) {
+  int n;
+  const char *k;
+  metric_id_t **udata, *metric_id;
+  n = lua_gettop(L); /* number of arguments */
+  assert(n == 2);
+  if(!luaL_checkudata(L, 1, "metric_id_t")) {
+    luaL_error(L, "metatable error, arg1 not a metric_id_t!");
+  }
+  udata = lua_touserdata(L, 1);
+  metric_id = *udata;
+  if(!lua_isstring(L, 2)) {
+    luaL_error(L, "metatable error, arg2 not a string!");
+  }
+
+  k = lua_tostring(L, 2);
+  switch (*k) {
+    case 'i':
+      if(!strcmp(k, "id")) {
+        char uuid_str[UUID_STR_LEN + 1];
```

Comment:
"string of the length defined in the uuid.h macro,
       UUID_PRINTABLE_STRING_LENGTH, which includes the trailing null
       character."   

Could use `UUID_PRINTABLE_STRING_LENGTH` instead of the +1